db/upgrade.php

```diff
@@ -64,6 +64,39 @@ function create_xapi_log_table($dbman, $tablename) {
     }
 }
 
+function create_logstoreid_and_type_columns($dbman) {
+
+    // Select table.
+    $table = new xmldb_table('logstore_xapi_log');
+
+    // Conditionally add field logstore_standard_log_id to logstore_xapi_log.
+    $field = new xmldb_field('logstore_standard_log_id', XMLDB_TYPE_INTEGER, '10', null, null, null, null);
+    if (!$dbman->field_exists($table, $field)) {
+        $dbman->add_field($table, $field);
+    }
+
+    // Conditionally add field type to logstore_xapi_log.
+    $field = new xmldb_field('type', XMLDB_TYPE_INTEGER, '10', null, null, null, null);
+    if (!$dbman->field_exists($table, $field)) {
+        $dbman->add_field($table, $field);
+    }
```

Comment:
this block is exactly the same as lines 87-97.

Wouldn't it better to rewrite as a `foreach` against an array containing the table names? 
If we change anything going forward, it means changing it once and eliminates the chance of a developer missing one of the tables/not duplicating it exactly.

Reply:
Or just put an extra parameter which  contains tablename in the function call.

Reply:
I have put this in a function and called it with a table name parameter.